app/models/post.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: posts
+#
+#  id               :bigint           not null, primary key
+#  author_id        :bigint           not null
+#  title            :string
+#  text             :text
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#  comments_counter :integer
+#  likes_counter    :integer
+#
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_user_posts_counter` method every time a new post has been saved to the database, then make the `update_user_posts_counter` method private

Reply:
Hi @DeliceLydia 
I really like this idea you share, thank you very much, I will implement this.